rfc/003-protocol.org

```diff
@@ -383,32 +383,77 @@ There are plans to re-design this protocol and use a binary
 serialisation for efficient processing; this section is intentionally
 brief.
 
-----
+** Replication
 
 Today this is a text-based protocol using a combination of
-tab-separated and comma-separated data fields. There is encoding for a
-reset control signal and a check-pointing mechanism - including acks
-from write-csv.
+tab-separated and comma-separated data fields. The payloads of the CSV
+values are encoded using a JSON representation, except for nulls which
+are the empty field.
+
+A payload can atomically update multiple rows across a set of tables.
+
+Here is a full example of an exchange:
+
+Sent to A:
+#+BEGIN_SRC text
+^table1 153
+0	"foo","bar","read-write"
+1	"foo","baz","read-write"
+^table2
+1	"some value",,4.3,7,"read-write"
+^table3 74
+!rebuild
+:197
+#+END_SRC
+
+This indicates that
+
+- we wish to update the 'foo' row of table1. We've seen up to tick 153
+  of A.
+- we wish to insert a row in to table2. We've not seen any data from
+  table 2 and so are leaving off the tick which is optional and
+  defaults to 0.
+- we cannot produce a diff for table3 (and we've seen up to A's tick
+  74). This happens due to tombstone GC. Without the tombs we cannot
+  reconstruct a full diff. Instead we communicate that the client
+  should rebuild their table by subscribing from 0 again and cold
+  starting.
+- this payload takes you up to date with the sender at tick 197. A
+  should update its watermarks appropriately.
+
+
+Response from A:
+#+BEGIN_SRC
+^table1__skdb_mirror_feedback
+:197
+^table2
```

Comment:
Should this be `table2__skdb_mirror_feedback`?

Reply:
Yes, great catch. Thanks.